kubetest2/kubetest2-kind/main.go

```diff
@@ -18,10 +18,12 @@ package main
 
 import (
 	"k8s.io/test-infra/kubetest2/pkg/app"
+	// import the standard set of testers so they are loaded & registered
+	_ "k8s.io/test-infra/kubetest2/pkg/app/testers/standard"
 
 	"k8s.io/test-infra/kubetest2/kubetest2-kind/deployer"
 )
 
 func main() {
-	app.Main(deployer.Name, deployer.New)
+	app.Main(deployer.Name, deployer.Usage(), deployer.New)
```

Comment:
This was a noisy change, we pass through `usage string` all throughout. This is because we want to have usage even without constructing a deployer.

HelpText for incorrect usage of a deployer or tester is still returned through an error type in `NewDeployer` / `NewTester`.

deployers can implement this when they have flags by:
- factor out flags to an options type
- init a copy of the options / flags in a `Usage()` func and return the flags usage
- use the same flag setup within `NewDeployer`

Reply:
have you considered `app.go::Main` to accept an interface or a options struct, or even the functional options pattern?

Reply:
I did, but I hope to not need to inject anything more. These deployer-binary `main.go`s are meant to be very simple and just allow injecting the definitions into the shared app.

We had to add Usage() here because originally this was returned during `NewDeployer` but that turned out to be impractical for the fixed usage text when actually implementing this (it still makes sense for errors with incorrect flags).

Reply:
Other things like the list of testers is injected by way of `import _ testers` and the `Register` in the testers package.

I expect this kind of change to settle down once we actually have a deployer or two figured out.

Reply:
ack.